projects/approved-premises-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/service/ApprovedPremisesService.kt

```diff
@@ -163,14 +166,38 @@ class ApprovedPremisesService(
         updateMainAddress(person, details)
     }
 
-    fun createAlertContact(
+    @Transactional
+    fun personDeparted(event: HmppsDomainEvent) {
+        val details = approvedPremisesApiClient.getPersonDepartedDetails(event.url()).eventDetails
+        val person = personRepository.getByCrn(event.crn())
+        val staff = staffRepository.getByCode(details.keyWorker.staffCode)
+        createContact(
+            person = person,
+            type = DEPARTED,
+            notes = "For details, see the referral on the AP Service: ${details.applicationUrl}",
+            date = details.departedAt,
+            staff = staff,
+            probationAreaCode = details.premises.probationArea.code,
+            createAlert = false
+        )
+        closeNsi(details)
+        endMainAddress(person, details.departedAt.toLocalDate())
+    }
+
+    private fun closeNsi(details: PersonDeparted) {
+        val nsi = nsiRepository.findByExternalReference(EXT_REF_BOOKING_PREFIX + details.bookingId)
```

Comment:
Should this also make the NSI inactive? Or is this done by the DB? if so can the active flag in the NSI entity remain a val?